### ISSUE TITLE
Different 'label' values are stored in the Graph database for package versions data

### ISSUE THREAD

**Issue by tisnik** [openshiftio/openshift.io#1934]
The 'label' values for package versions seems not to be consistent, at least on the __Stage__ Graph database. Sometimes the value is set simply to `"vertex"` and sometimes to `"Version"`.

Reproducer:

```
import os
import sys
import requests

URL=STAGE_GREMLIN_URL


def gremlin_search_package_in_ecosystem(ecosystem, package):
    query = 'g.V().has("ecosystem", "{ecosystem}").has("name", "{package}").out("has_version")'.format(ecosystem=ecosystem, package=package)
    print(query)
    data = post_query(query)
    try:
        print("*** " + package + " ***")
        assert data["result"]["data"] is not None
        for e in data["result"]["data"]:
            print(e["label"])
        print()
    except Exception as e:
        print("none")


def post_query(query):
    data = {"gremlin": str(query)}
    response = requests.post(URL, json=data)
    #print(response.status_code)
    data = response.json()
    return data


packages = [
    "sequence",
    "array-differ"
]

for package in packages:
    gremlin_search_package_in_ecosystem("npm", package)

```

Actual output:

```
g.V().has("ecosystem", "npm").has("name", "sequence").out("has_version")
*** sequence ***
vertex
vertex

g.V().has("ecosystem", "npm").has("name", "array-differ").out("has_version")
*** array-differ ***
Version
Version
Version
Version
Version
```

@msrb @pkajaba FYI

**Comment by qodfathr** [comment]
@tisnik can you please add a severity level? Thank you.

**Comment by tuxdna** [comment]
I setup a local instance and ingested these two packages:
```
$ ./docker-compose.sh up

# in a separate terminal
$ wget http://localhost:32000/api/v1/component-analyses/npm/array-differ/2.0.3
$ wget http://localhost:32000/api/v1/component-analyses/npm/sequence/3.0.0
```

With these freshly ingested npm packages on my instance I get:

```
$ python3 script.py 
g.V().has("ecosystem", "npm").has("name", "sequence").out("has_version")
*** sequence ***
Node type: vertex
Property vertex_label: Version

g.V().has("ecosystem", "npm").has("name", "array-differ").out("has_version")
*** array-differ ***
Node type: vertex
Property vertex_label: Version
                               
```
So it is clear that the labels are set correctly in current implementation, using **Node properties** rather than node label.

I could be that data in Stage is from a very early run of the ingestion that was buggy as some point.

**Comment by tuxdna** [comment]
@tisnik how did you access the stage graph instance ?

**Comment by tuxdna** [comment]
Marking as blocked until we have access to Graph instance on production ( accessible behind an auth layer )

**Comment by miteshvp** [comment]
@tisnik good catch. To me it seems like a non-issue since we use `vertex_label` and not `label` property. BTW, the reason you see different label props is because initially when we ingested the data, we used to apply `label` and `vertex_label` the same value [More Details](https://github.com/fabric8-analytics/fabric8-analytics-data-model/commit/c2a4948fcf27dc180a9c01118a228be67a3ed03f#diff-97e02c85ab62bb7f4928ef1ebc1a7487R216). As we progressed, we just apply `vertex_label` property that we rely on and let Titan/Gremlin use the default `label` property. Hence data ingested earlier have `label` value as `Version` and later on, it took the default value provided which is `vertex`. Having said that, it should not have any issues with its functioning.
Feel free to close this issue if you agree.
